mmv1/templates/terraform/examples/privateca_certificate_generated_key.tf.erb

```diff
@@ -0,0 +1,91 @@
+# [START privateca_create_certificate_config]
+
+resource "google_privateca_ca_pool" "default" {
+  location = "us-central1"
+  name = "default"
+  tier = "ENTERPRISE"
+}
+
+resource "google_privateca_certificate_authority" "default" {
+  location = "us-central1"
+  pool = google_privateca_ca_pool.default.name
+  certificate_authority_id = "my-authority"
+  config {
+    subject_config {
+      subject {
+        organization = "HashiCorp"
+        common_name = "my-certificate-authority"
+      }
+      subject_alt_name {
+        dns_names = ["hashicorp.com"]
+      }
+    }
+    x509_config {
+      ca_options {
+        is_ca = true
+      }
+      key_usage {
+        base_key_usage {
+          cert_sign = true
+          crl_sign = true
+        }
+        extended_key_usage {
+          server_auth = true
+        }
+      }
+    }
+  }
+  key_spec {
+    algorithm = "RSA_PKCS1_4096_SHA256"
+  }
+
+  // Disable CA deletion related safe checks for easier cleanup.
+  deletion_protection                    = false
+  skip_grace_period                      = true
+  ignore_active_certificates_on_deletion = true
+}
+
+resource "tls_private_key" "cert_key" {
+  algorithm = "RSA"
+}
+
+resource "google_privateca_certificate" "<%= ctx[:primary_resource_id] %>" {
+  location = "us-central1"
+  pool = google_privateca_ca_pool.default.name
+  lifetime = "86000s"
+  name = "cert-1"
+  config {
+    subject_config  {
+      subject {
+        common_name = "san1.example.com"
+        country_code = "us"
+        organization = "google"
+        organizational_unit = "enterprise"
+        locality = "mountain view"
+        province = "california"
+        street_address = "1600 amphitheatre parkway"
+      } 
+      subject_alt_name {
+        dns_names = ["example.google.com"]
+      }
+    }
+    x509_config {
+      ca_options {
+        is_ca = false
+      }
+      key_usage {
+        base_key_usage {
+          digital_signature = true
+        }
+        extended_key_usage {
+          server_auth = false
+        }
+      }
+    }
+    public_key {
+      format = "PEM"
+      key = base64encode(tls_private_key.cert_key.public_key_pem)
```

Comment:
Since we're not running this as an automated test, I just wanted to double-check that you've been able to run this configuration locally and it works fine?

Reply:
Yes. Tested locally.